### ISSUE TITLE
[bug] HCM: Fix hover and focus color for `post-avatar` in anchor and button contexts

### ISSUE THREAD

**Issue by alionazherdetska** [swisspost/design-system#4294]
**Description:**
The `post-avatar` component is missing proper hover and focus color behaviors in specific contexts:

1. When nested inside an anchor tag:
    The hover color is not applied.
2. When inside a button:
    The hover effect is missing.
    The focus color is applied incorrectly, appearing on top of the avatar border.
 ![Image](https://github.com/user-attachments/assets/fe6f6746-9846-44b9-9133-2b644ebbe301)

**Steps to Reproduce:**

1. Enable high contrast mode in the browser or system settings.
2. Navigate to a page for `post-avatar` component.
3. Hover over or focus on the <post-avatar> component nested inside the anchor tag and a button.
4. Observe the missing or inconsistent hover and focus styles.
